.gitignore

```diff
@@ -1,6 +1,7 @@
 /CMakeFiles
 /build*/*
 /examples/hello_world/build
+/.idea/
```

Comment:
Can you also add a few other common IDE/build artifacts?  I have a PR for this too but it's still in a very early draft state.
```
/.cache
/compile_commands.json
.ninja
Makefile
```